### PR TITLE
[SM] Fill in search fields with query params after successfully loading folder.

### PR DESCRIPTION
**Pull request by U-DON**
Closes department-of-veterans-affairs/messaging-fe#149.

### CODE REVIEW

**Comment by U-DON** [comment]
I left a little code in the test for filtering on recipient, which isn't relevant yet, but will be after I get the code in for showing a 'To' field instead of 'From' for searching in Drafts and Sent.

File: src/js/messaging/reducers/search.js [hunk 1]
```diff
@@ -34,6 +35,40 @@ const initialState = {
 
 export default function modals(state = initialState, action) {
   switch (action.type) {
+    case FETCH_FOLDER_SUCCESS: {
+      const { filter } = action.messages.meta;
+      if (!filter) { return state; }
+
+      const params = {
+        dateRange: { ...initialState.params.dateRange },
```
Comment (bshyong) [inline src/js/messaging/reducers/search.js]: Would this be the same as `dateRange: initialState.paras.dateRange`?

**Review by bshyong — APPROVED** [review]
LGTM /w a minor question

File: src/js/messaging/reducers/search.js (hunk 1 above)
Comment (U-DON) [inline src/js/messaging/reducers/search.js]: I just wanted a deep enough clone of `initialState`. If it were `dateRange: initialState.params.dateRange`, I think setting `params.dateRange.start = newDate` would affect `initialState.dateRange`.

Comment (bshyong) [inline src/js/messaging/reducers/search.js]: :+1:

**Pull request merged**